README.md

```diff
@@ -1,2 +1,941 @@
+```
+
+`navigateSiteSameTab` is a method that given a `url`, it'll **navigate** to that site on a **new tab**.
+
+```typescript
+navigateSiteNewTab(url: string): void;
+```
+
+`navigateSitePostSameTab` is a method that given a `url`, it'll **navigate** to that site on the **same tab**. It'll also take in an `value` and does a `POST` on that site.
+
+```typescript
+navigateSitePostSameTab(url: string, value: object): void;
+```
+
+`navigateSitePostNewTab` is a method that given a `url`, it'll **navigate** to that site on the **new tab**. It'll also take in an `value` and does a `POST` on that site.
+
+```typescript
+navigateSitePostNewTab(url: string, value: object): void;
+```
+
+`getSiteQueryParam` is a method that given a `key`, returns the **query parameter** for the site.
+
+```typescript
+getSiteQueryParam(key: string): string;
+```
+
+<a id="examples"></a>
+
+## **Examples**
+
+<a id="simple-counter"></a>
+
+### **Simple Counter**
+
+Here's an example of an simple counter.  
+
+Let's create a UI for our Simple Counter. Start by creating a HTML **view** file by naming `counter-view.html` and adding a button which a click event that calls `incrementByOne` method and display it inside a `p` tag:
+
+```html
+<div class="counter">
+    <button @click="incrementByOne">Add One</button>
+    <p class="is-blue">{{ count }}</p>
+</div>
+```
+
+Now let's make the `p` tag blue by creating a SCSS **view** file naming it `counter-view.scss`:
+
+```scss
+.counter
+{
+    .is-blue
+    {
+        color: blue;
+    }
+}
+```
+
+We need some functionality. Let's create a TypeScript **view-model** file naming it `counter-view-model.ts` and importing **n-app** into our application and making our view-model class a **subclass** of the `PageViewModel`. We'll also route our page and add a method called `incrementByOne` that increments a private number property called `_count` by 1 and we'll expose that property using a **getter** which will then be displayed to the **view**:
+
+```typescript
+import { PageViewModel, route, template } from "@nivinjoseph/n-app";
+import { Routes } from "../routes";
+import "./counter-view.scss";
+
+@template(require("./counter-view.html")) 
+@route(Routes.counter)
+export class CounterViewModel extends PageViewModel 
+{
+    private _count = 0;
+
+    public get count(): number { return this._count; }
+
+    public incrementByOne(): void
+    {
+        this._count++;
+    }
+}
+```
+
+Now let's link our previous **route** by creating a `Routes` class which contains the `counter` route. Let's name this file `routes.ts`:
+
+```typescript
+export class Routes
+{
+    public static readonly counter = "/counter";
+
+    public static readonly default = Routes.counter;
+}
+```
+
+Now in our `index.ts`, we'll import our **View-Model**:
+
+```typescript
+import { CounterViewModel } from "./counter-view-model.ts";
+
+export const pages = [
+    CounterViewModel
+];
+```
+
+Now we're done!
```

Comment:
No, you are not done yet, you forgot the most important part, the client.ts file where everything comes together